src/main.c

```diff
@@ -10,25 +12,80 @@
 #include "emit.h"
 #include "mapper.h"
 
+static int should_reload = 0;
+static int should_exit = 0;
+
+/**
+ * Handles signal events.
+ * */
+static void on_signal(int signal)
+{
+    if (signal == SIGHUP)
+    {
+        should_reload = 1;
+    }
+    else if (signal == SIGINT || signal == SIGTERM)
+    {
+        should_exit = 1;
+    }
+    else if (signal == SIGSEGV)

```

Comment:
Why catch SIGSEGV, do nothing and not properly exit with EXIT_FAILURE?

Reply:
I don't believe there's anything useful I can do and hiding the segmentation fault with EXIT_FAILURE isn't ideal. I'm removing the handler.